admin/services/generate-pins.service.js

```diff
@@ -14,7 +18,7 @@ const generatePinService = {
     let pupils = await pupilDataService.getSortedPupils(schoolId, sortField, sortDirection)
     // filter pupils
     pupils = pupils
-      .filter(p => !p.pin && !p.attendanceCode && !p.result)
+      .filter(p => !generatePinService.isValidPin(p.pin, p.pinExpiresAt) && !p.attendanceCode && !p.result)
```

Comment:
This whole line is a bit much. I would suggest moving this to a function - removePupilsWithPin or something

Reply:
@jon-shipley updated